ext/drcovlib/modules.c

```diff
@@ -174,11 +176,12 @@ event_module_load(void *drcontext, const module_data_t *data, bool loaded)
                 }
             }
 #endif
-            break;
+            /* We may just see a sub module, so we cannot stop here. We need iterate
+             * all entries to update all sub modules of the reloaded module.
```

Comment:
I don't understand: "We may just see a sub module": you mean if the app does a custom mmap of just one segment of an ELF file, and then goes and executes from that segment to trigger the module load event?  But why would that require further iteration?  All non-contiguous entries are guaranteed to be in the vector immediately following the initial entry.  Why would we need to iterate further?